File: client/src/lib/invoke/getGitlabOauthToken.ts
import { invoke } from "@tauri-apps/api/core";
import { ResultAsync } from "neverthrow";

export function getGitlabOauthToken(): ResultAsync<string, string> {
    return ResultAsync.fromPromise(
        invoke("get_gitlab_oauth_token"),
        (e) => `Failed to get gitlab oauth token: ${e}`
    );
}

File: client/src-tauri/src/commands/get_gitlab_oauth_token.rs
use nexium::gitlab::GitlabClient;

#[tauri::command]
pub async fn get_gitlab_oauth_token() -> Result<String, String> {
    tauri::async_runtime::spawn_blocking(move || {
        GitlabClient::get_token().map_err(|err| err.to_string())
    })
    .await
    .map_err(|err| format!("Failed to get gitlab oauth token: {}", err))?
}
